### ISSUE TITLE
White space between DOM elements in chat messes with the visuals and copy/pasting

### ISSUE THREAD

**Issue by AlcaDesign** [night/BetterTTV#2179]
Image:
![image](https://cloud.githubusercontent.com/assets/7132646/24266629/7427d598-0fd5-11e7-9cdf-dc4abe4575fd.png)

Copy/paste:
```
2:27  Moderator  2-Year Subscriber  Turbo  Alca :  LUL
2:27  Moderator  2-Year Subscriber  Turbo  Alca :  Jebaited
```

Original DOM:
```
<li id="ember5318" class="message-line chat-line ember-view"><!---->  <span class="timestamp float-left">2:27</span>
<!---->    <span class="mod-icons float-left">
        <span class="balloon-wrapper float-left">
          <a title="Ban User" href="#" class="mod-icon float-left ban" data-ember-action="" data-ember-action-5319="5319">Ban</a>
          <div class="balloon balloon--tooltip balloon--up balloon--center">Ban</div>
        </span>
      <span class="balloon-wrapper float-left">
        <a title="Timeout User" href="#" class="mod-icon float-left timeout" data-ember-action="" data-ember-action-5320="5320">Timeout</a>
        <div class="balloon balloon--tooltip balloon--up balloon--center">Timeout</div>
      </span>
    </span>
  <span class="badges float-left">
      <span class="balloon-wrapper float-left">
        <a rel="noopener noreferrer" target="_blank">
          <img src="https://static-cdn.jtvnw.net/badges/v1/3267646d-33f0-4b17-b3df-f923a41db1d0/1" srcset="https://static-cdn.jtvnw.net/badges/v1/3267646d-33f0-4b17-b3df-f923a41db1d0/2 2x, https://static-cdn.jtvnw.net/badges/v1/3267646d-33f0-4b17-b3df-f923a41db1d0/3 4x" alt="Moderator" class="badge">
        </a>
        <div class="balloon balloon--tooltip balloon--up">Moderator</div>
      </span>
      <span class="balloon-wrapper float-left">
        <a href="/dome/subscribe?ref=in_chat_subscriber_link" rel="noopener noreferrer" target="_blank">
          <img src="https://static-cdn.jtvnw.net/badges/v1/69d710bb-7161-436a-a0a7-f45da6ef568d/1" srcset="https://static-cdn.jtvnw.net/badges/v1/69d710bb-7161-436a-a0a7-f45da6ef568d/2 2x, https://static-cdn.jtvnw.net/badges/v1/69d710bb-7161-436a-a0a7-f45da6ef568d/3 4x" alt="2-Year Subscriber" class="badge">
        </a>
        <div class="balloon balloon--tooltip balloon--up">2-Year Subscriber</div>
      </span>
      <span class="balloon-wrapper float-left">
        <a href="/products/turbo?ref=chat_badge" rel="noopener noreferrer" target="_blank">
          <img src="https://static-cdn.jtvnw.net/badges/v1/bd444ec6-8f34-4bf9-91f4-af1e3428d80f/1" srcset="https://static-cdn.jtvnw.net/badges/v1/bd444ec6-8f34-4bf9-91f4-af1e3428d80f/2 2x, https://static-cdn.jtvnw.net/badges/v1/bd444ec6-8f34-4bf9-91f4-af1e3428d80f/3 4x" alt="Turbo" class="badge">
        </a>
        <div class="balloon balloon--tooltip balloon--up">Turbo</div>
      </span>
  </span>
  <span style="color: rgb(65, 150, 235);" class="from" data-ember-action="" data-ember-action-5324="5324">Alca</span>
  <span class="colon">:</span>
      <span style="" class="message">
            <span>
            <span class="balloon-wrapper">
                <img src="//cdn.betterttv.net/emote/567b00c61ddbe1786688a633/1x" srcset="//cdn.betterttv.net/emote/567b00c61ddbe1786688a633/2x 2x, //cdn.betterttv.net/emote/567b00c61ddbe1786688a633/3x 4x" alt="LUL" class="emoticon bttv bttv-emo-567b00c61ddbe1786688a633">
                <div class="balloon balloon--tooltip balloon--up balloon--center mg-t-1" style="text-align: center;">
            LUL<br>
            
            BetterTTV Global Emotes
        </div>
            </span>
         </span>
                </span>
<!----><!----></li>
```
```
<li id="ember5328" class="message-line chat-line ember-view"><!---->  <span class="timestamp float-left">2:27</span>
<!---->    <span class="mod-icons float-left">
        <span class="balloon-wrapper float-left">
          <a title="Ban User" href="#" class="mod-icon float-left ban" data-ember-action="" data-ember-action-5329="5329">Ban</a>
          <div class="balloon balloon--tooltip balloon--up balloon--center">Ban</div>
        </span>
      <span class="balloon-wrapper float-left">
        <a title="Timeout User" href="#" class="mod-icon float-left timeout" data-ember-action="" data-ember-action-5330="5330">Timeout</a>
        <div class="balloon balloon--tooltip balloon--up balloon--center">Timeout</div>
      </span>
    </span>
  <span class="badges float-left">
      <span class="balloon-wrapper float-left">
        <a rel="noopener noreferrer" target="_blank">
          <img src="https://static-cdn.jtvnw.net/badges/v1/3267646d-33f0-4b17-b3df-f923a41db1d0/1" srcset="https://static-cdn.jtvnw.net/badges/v1/3267646d-33f0-4b17-b3df-f923a41db1d0/2 2x, https://static-cdn.jtvnw.net/badges/v1/3267646d-33f0-4b17-b3df-f923a41db1d0/3 4x" alt="Moderator" class="badge">
        </a>
        <div class="balloon balloon--tooltip balloon--up">Moderator</div>
      </span>
      <span class="balloon-wrapper float-left">
        <a href="/dome/subscribe?ref=in_chat_subscriber_link" rel="noopener noreferrer" target="_blank">
          <img src="https://static-cdn.jtvnw.net/badges/v1/69d710bb-7161-436a-a0a7-f45da6ef568d/1" srcset="https://static-cdn.jtvnw.net/badges/v1/69d710bb-7161-436a-a0a7-f45da6ef568d/2 2x, https://static-cdn.jtvnw.net/badges/v1/69d710bb-7161-436a-a0a7-f45da6ef568d/3 4x" alt="2-Year Subscriber" class="badge">
        </a>
        <div class="balloon balloon--tooltip balloon--up">2-Year Subscriber</div>
      </span>
      <span class="balloon-wrapper float-left">
        <a href="/products/turbo?ref=chat_badge" rel="noopener noreferrer" target="_blank">
          <img src="https://static-cdn.jtvnw.net/badges/v1/bd444ec6-8f34-4bf9-91f4-af1e3428d80f/1" srcset="https://static-cdn.jtvnw.net/badges/v1/bd444ec6-8f34-4bf9-91f4-af1e3428d80f/2 2x, https://static-cdn.jtvnw.net/badges/v1/bd444ec6-8f34-4bf9-91f4-af1e3428d80f/3 4x" alt="Turbo" class="badge">
        </a>
        <div class="balloon balloon--tooltip balloon--up">Turbo</div>
      </span>
  </span>
  <span style="color: rgb(65, 150, 235);" class="from" data-ember-action="" data-ember-action-5334="5334">Alca</span>
  <span class="colon">:</span>
      <span style="" class="message">
            <span class="balloon-wrapper">
              <img src="//static-cdn.jtvnw.net/emoticons/v1/114836/1.0" srcset="//static-cdn.jtvnw.net/emoticons/v1/114836/2.0 2x" alt="Jebaited" class="emoticon">
              <div class="balloon balloon--tooltip balloon--up balloon--center mg-t-1">Jebaited</div>
            </span>
             
                </span>
<!----><!----></li>
```

**Comment by night** [comment]
this is default twitch behavior, we do not have control over this anymore, sorry.

**Comment by AlcaDesign** [comment]
Why? ![DansGame](https://static-cdn.jtvnw.net/emoticons/v1/33/1.0)

**Comment by night** [comment]
we do not handle rendering of messages anymore, twitch does it, and then we modify the dom a bit.. it's more expensive to modify the dom after creation, so we will not be modifying entire chat line structure

**Comment by EhsanKia** [comment]
Visual can be fixed easily with CSS (will do that tonight, it's always bugged me in Conversations/Replay and now it's in normal chat too). I don't see why copy pasting really matters that much.

**Comment by AlcaDesign** [comment]
I was doin' my moderator thing in a chat and I needed to copy/paste a command from someone to add to Nightbot. It was a "streamer shout out" type of command, so there's a link. In essence, what I copied was like this: "Check out $(touser)! twitch.tv/ $(touser)" which would result in an issue because there's a space in the link. Plus, there's no up-arrow chat history in the Beta so turned out to be a fuckin' fiasco.

Also seeing a space before **and** after the colon in a chat message really irks me. No one _wants_ that.

`Alca : message here`
